src/odemis/gui/cont/tabs.py

```diff
@@ -2417,48 +2472,151 @@ def _update_progress_bar(self, pos):
         # Set the move gauge with the movement progress percentage
         self.panel.gauge_move.Value = val * 100
 
+    @call_in_wx_main
+    def _on_stage_pos(self, pos):
+        if self._role == 'meteor':
+            current_pos_label = getCurrentPositionLabel(pos)
+            if current_pos_label == LOADING:
+                self.panel.btn_switch_sem_imaging.Enable()
+                self.panel.btn_switch_fm_imaging.Enable()
+                self.panel.btn_switch_grid2.Disable()
+                self.panel.btn_switch_grid1.Disable()
+            elif current_pos_label != self._current_position and current_pos_label != UNKNOWN:
+                current_grid_label = getCurrentGridLabel(pos)
+                if current_grid_label == GRID_1:
+                    self.position_btns[GRID_1].SetValue(True)
+                    self.position_btns[GRID_2].SetValue(False)
+                elif current_grid_label == GRID_2:
+                    self.position_btns[GRID_2].SetValue(True)
+                    self.position_btns[GRID_1].SetValue(False)
+                if current_pos_label == SEM_IMAGING:
+                    self.position_btns[SEM_IMAGING].SetValue(True)
+                    self.position_btns[FM_IMAGING].SetValue(False)
+                elif current_pos_label == FM_IMAGING:
+                    self.position_btns[SEM_IMAGING].SetValue(False)
+                    self.position_btns[FM_IMAGING].SetValue(True)
+                self._enable_movement_controls()
+        elif self._role == 'enzel':
+            pass 
+
+
     def _toggle_switch_buttons(self, currently_pressed=None):
         """
         Toggle currently pressed button (if any) and untoggle rest of switch buttons
         """
-        for button in self.position_btns.values():
-            button.SetValue(1) if button == currently_pressed else button.SetValue(0)
+        if self._role == 'enzel':
+            for button in self.position_btns.values():
+                button.SetValue(1) if button == currently_pressed else button.SetValue(0)
+        elif self._role == "meteor":
+            # inital case 
+            if self.position_btns[GRID_1].GetValue() == 0 and self.position_btns[GRID_2].GetValue() == 0:
```

Comment:
This is a really complex flow. I think it'd be clearer if you had a separate function for Grid buttons. These buttons are mostly independent from the "position" buttons.

Reply:
First of all,  i think it is not accurate to say that the grid buttons are independent of the position buttons. They clearly have some dependency on each other. 
For making separate functions for the grid buttons, I have looked into that. It actually will not make the code simpler. I would have to copy paste the code existing in the current button callback to the new grid buttons' callback. So that would literally be repeated code. Additionally, it will not change this code above. We need to distinguish between the initial case and the normal, and handle them separately. However, i do think that this code of the toggle function was not written in a readable way. I refactored the code. So now it is a bit simpler.